services/web/server/tests/unit/with_dbs/01/test_api_keys.py

```diff
@@ -144,54 +143,66 @@ async def test_delete_api_keys(
         await assert_status(resp, expected)
 
 
+EXPIRATION_WAIT_FACTOR = 1.2
+
+
 @pytest.mark.parametrize(
     "user_role,expected",
     _get_user_access_parametrizations(status.HTTP_200_OK),
 )
 async def test_create_api_key_with_expiration(
+    disabled_setup_garbage_collector: MockType,
     client: TestClient,
     logged_user: UserInfoDict,
     user_role: UserRole,
     expected: HTTPStatus,
-    disable_gc_manual_guest_users: None,
+    mocker: MockerFixture,
 ):
     assert client.app
 
+    # test gc is actually disabled
+    gc_prune_mock = mocker.patch(
+        "simcore_service_webserver.garbage_collector._tasks_api_keys.create_background_task_to_prune_api_keys",
+        spec=True,
+    )
+    assert not gc_prune_mock.called
+
+    expected_api_key = "foo"
+
     # create api-keys with expiration interval
     expiration_interval = timedelta(seconds=1)
     resp = await client.post(
         "/v0/auth/api-keys",
-        json={"displayName": "foo", "expiration": expiration_interval.seconds},
+        json={
+            "displayName": expected_api_key,
+            "expiration": expiration_interval.seconds,
+        },
     )
 
     data, errors = await assert_status(resp, expected)
     if not errors:
-        assert data["displayName"] == "foo"
+        assert data["displayName"] == expected_api_key
         assert "apiKey" in data
         assert "apiSecret" in data
 
         # list created api-key
         resp = await client.get("/v0/auth/api-keys")
         data, _ = await assert_status(resp, expected)
-        assert [d["displayName"] for d in data] == ["foo"]
+        assert [d["displayName"] for d in data] == [expected_api_key]
 
         # wait for api-key for it to expire and force-run scheduled task
-        async for attempt in tenacity.AsyncRetrying(
-            wait=wait_fixed(1),
-            retry=retry_if_exception_type(AssertionError),
-            stop=stop_after_delay(5 * expiration_interval.seconds),
-            reraise=True,
-        ):
-            with attempt:
-                deleted = await prune_expired_api_keys(client.app)
-                assert deleted == ["foo"]
+        await asyncio.sleep(EXPIRATION_WAIT_FACTOR * expiration_interval.seconds)
```

Comment:
Q: why was the tenacity removed from here and now we have a sleep? Are you sure it's better. Usually we have to put in the retry pattern to avoid flaky tests

Reply:
@GitHK as I explained in the description, this test was flaky because the background task was **not disabled**. Here tenacity does not really play a role since we explicilty call the background function `prune_expired_api_keys` . The only reason we wait is for the api-key to be expired, not to check that the background function ran.